lib/resty/mlcache/ipc.lua

```diff
@@ -155,23 +171,68 @@ function _M:poll()
     for _ = self.idx, idx - 1 do
         self.idx = self.idx + 1
 
-        local v, err = self.dict:get(self.idx)
-        if err then
-            return nil, "failed to get event from shm: " .. err
-        end
+        -- fetch event from shm with a retry policy in case
+        -- we run our :get() in between another worker's
+        -- :incr() and :set()
+        -- 1: 0.001s
+        -- 2: 0.010s
+        -- 3: 0.100s
+        -- 4: 1.000s
+
+        local v
+
+        do
+            local perr
+            local pok         = true
+            local retry_delay = 0.0001
+
+            if not max_sleep then
+                max_sleep = 1
+            end
 
-        if not v then
-            return nil, "no event at index: " .. idx
+            while retry_delay < max_sleep do
```

Comment:
I think not missing an event is of critical importance in the case of poll(), as it is the promise that poll() will make a worker consistent with all the other ones. As such, I think 10x is fine, eventually, 0.1 could be the max amount of time allowed for a sleep, as 1s might be overkill. (this is the use-case for `max_sleep`, which I agree does not have a good enough name, or usage indeed.